### ISSUE TITLE
Empty migration results in total cost being printed as `0 undefined`

### ISSUE THREAD

**Issue by haltman-at** [trufflesuite/truffle#4408]
## Issue

Suppose you have a migrations file, but no actual migrations occur in it.  When you run `truffle migrate`, the total cost is printed as `0 undefined`, rather than `0 wei` or `0 ether` or just `0`.

## Steps to Reproduce

Create a new Truffle project, edit `1_initial_migrations.js` by removing the `deployer.deploy` line, then try migrating your project.

## Expected Behavior

The 0 costs should not have "undefined" displayed as their unit.

## Actual Results

```
Starting migrations...
======================
> Network name:    'development'
> Network id:      1636404659203
> Block gas limit: 12000000 (0xb71b00)


1_initial_migration.js
======================
   -------------------------------------
   > Total cost:                   0 undefined


Summary
=======
> Total deployments:   0
> Final cost:          0 undefined
```

## Environment

* Truffle version (`truffle version`): 5.4.18

**Comment by haltman-at** [comment]
Oh hey @lsqproduction, process note, typically we don't close issues on the Truffle repo until the fix is released, rather than as soon as it's merged.  If you don't mind, I'm going to reopen this one so it can be closed properly when the release is done. :)

**Comment by gnidan** [comment]
This is half-done and now released in [Truffle v5.4.20](https://github.com/trufflesuite/truffle/releases/tag/v5.4.20).

(Leaving this open because it's fixed for individual migrations but not for the total at the bottom)

cc @lsqproduction

**Comment by hughes-ch** [comment]
Should be fixed with pull request #4684. I've also added a test to reproduce (and prevent regressions)

**Comment by cds-amal** [comment]
This should be resolved in the latest [release](https://github.com/trufflesuite/truffle/releases/tag/v5.4.32)
